src/Azure.Functions.Cli/Actions/HostActions/StartHostAction.cs

```diff
@@ -418,15 +431,36 @@ private async Task<JObject> GetLocalSettingsJsonAsJObjectAsync()
         public override async Task RunAsync()
         {
             await PreRunConditions();
+            var isVerbose = VerboseLogging.HasValue && VerboseLogging.Value;
 
-            var isCurrentProcessNet6Build = RuntimeInformation.FrameworkDescription.Contains(Net6FrameworkDescriptionPrefix);
-            if (isCurrentProcessNet6Build && ShouldLaunchInProcNet8AsChildProcess() && await IsInProcNet8Enabled())
+            var isCurrentProcessNet8Build = RuntimeInformation.FrameworkDescription.Contains(Net8FrameworkDescriptionPrefix);
+            // If --runtime param is set, handle runtime param logic; otherwise we infer the host to launch on startup
+            if (HostRuntime != null)
             {
-                await StartInProc8AsChildProcessAsync();
-                return;
+                // Check if we should start child process from user specified host runtime and return
+                var shouldStartChildProcess = await ShouldStartChildProcessFromHostRuntime(isCurrentProcessNet8Build, isVerbose);
+
+                if (shouldStartChildProcess)
+                {
+                    var isNet8InProcSpecified = (string.Equals(HostRuntime, InProc8HostRuntime, StringComparison.OrdinalIgnoreCase)) ? true : false;
```

Comment:
`string.Equals` already returns a bool so no need to do a conditional check here

```suggestion
                    var isNet8InProcSpecified = string.Equals(HostRuntime, InProc8HostRuntime, StringComparison.OrdinalIgnoreCase);
```